src/backend/joanie/core/api/client.py

```diff
@@ -447,6 +468,35 @@ def invoice(self, request, pk=None):  # pylint: disable=no-self-use, invalid-nam
 
         return response
 
+    @action(detail=True, methods=["PUT"])
+    def add_payment(
+        self, request, pk=None
+    ):  # pylint: disable=no-self-use, invalid-name
+        """
+        Create a payment for an order
+        """
+        serializer = self.get_serializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        billing_address = serializer.initial_data.get("billing_address")
+        if not billing_address:
+            return Response(
+                {"billing_address": ["This field is required."]}, status=400
+            )
+        order = models.Order.objects.get(pk=pk)
+        payment_info = self._create_payment(order, serializer, request, billing_address)
+        return Response({"payment_info": payment_info}, status=201)
+
```

Comment:
We don't need an action for this. The client creates the order, fills in all the necessary on the order via normal updates, then calls the submit action. The payment is then created in the fsm transitions.